copy/single.go

```diff
@@ -365,26 +385,34 @@ func (ic *imageCopier) compareImageDestinationManifestEqual(ctx context.Context,
 		return nil, nil
 	}
 
+	compressionAlgos := []compressiontypes.Algorithm{}
+	for _, srcInfo := range ic.src.LayerInfos() {
+		compression := ic.getCompressionFromBlob(srcInfo)
+		compressionAlgos = append(compressionAlgos, *compression)
+	}
+
 	// Destination and source manifests, types and digests should all be equivalent
 	return &copySingleImageResult{
 		manifest:         destManifest,
 		manifestMIMEType: destManifestType,
 		manifestDigest:   srcManifestDigest,
+		compressionAlgorithms: removeDuplicateCompression(compressionAlgos),
 	}, nil
 }
 
 // copyLayers copies layers from ic.src/ic.c.rawSource to dest, using and updating ic.manifestUpdates if necessary and ic.cannotModifyManifestReason == "".
-func (ic *imageCopier) copyLayers(ctx context.Context) error {
+func (ic *imageCopier) copyLayers(ctx context.Context) ([]compressiontypes.Algorithm, error) {
+	res := []compressiontypes.Algorithm{}
 	srcInfos := ic.src.LayerInfos()
 	numLayers := len(srcInfos)
 	updatedSrcInfos, err := ic.src.LayerInfosForCopy(ctx)
 	if err != nil {
-		return err
+		return res, err
```

Comment:
Non-blocking: I’d prefer a `return nil, …` on error paths, so that there is no data dependency and no ambiguity about what we return.